.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,177 @@
+        id: restore-test-durations
+        uses: actions/cache/restore@v4
+        with:
+          path: test_durations_merged.json
+          key: ${{ env.DURATIONS_CACHE_NAME }}
+
+      - name: Rename test duration artifacts
+        if: steps.restore-test-durations.outputs.cache-hit == 'true'
+        run: mv test_durations_merged.json .test_durations
+
+      - name: Set up Pyodide virtual environment and test SymPy for Pyodide
+        run: |
+          # Set up Pyodide virtual environment
+          pyodide xbuildenv install ${{ env.PYODIDE_VERSION }}
+          pyodide venv .venv-pyodide
+
+          # Activate the virtual environment and install SymPy
+          source .venv-pyodide/bin/activate
+          pip install .
+
+          # Install dependencies and optional test dependencies
+          pip install pytest pytest-split hypothesis
+          pip install matplotlib numpy scipy
+
+          # Trigger test suite (slow and tooslow tests are skipped by default)
+          pytest -svra --pyargs sympy \
+          --durations 20 \
+          --splits ${{ matrix.splits }} --group ${{ matrix.group }} \
+          --store-durations
+
+      - name: Move and rename test duration artifacts
+        if: always()
+        run: mv .test_durations test_durations_group_${{ matrix.group }}.json
+
+      - name: Upload test duration artifacts
+        if: always()
+        uses: actions/upload-artifact@v4.4.3
+        with:
+          name: pytest_durations_${{ matrix.group }}
+          path: test_durations_group_${{ matrix.group }}.json
+          if-no-files-found: error
+          # dotfiles are hidden by default, so we rename them above
+          # post actions/upload-artifact@v4.4.0 to make them visible
+          # and to avoid needing the `include-hidden-files` option
+
+  # We delete the previous cache to avoid storing outdated test durations.
+  # This is currently not possible to do with the actions/cache action:
+  # https://github.com/actions/toolkit/issues/505
+  # https://github.com/actions/cache/issues/342
+  #
+  # so we use the gh cache CLI to get around this limitation and delete
+  # any previous test durations before saving new ones. This exists
+  # as a separate job in order to scope run permissions differently
+  # from other jobs for security reasons.
+  delete-test-durations-caches:
+    name: Delete previous test duration caches
+    runs-on: ubuntu-latest
+    needs: [pyodide-test]
+    permissions:
+      contents: read
+      actions: write # to allow deletion of caches
+    if: always() && github.event_name == 'pull_request'
+    steps:
+      # Don't fail if the cache doesn't exist or if there are
+      # insufficient permissions to delete it. his is a best-effort
+      # attempt to delete the cache.
+      #
+      # This will fail for PRs from forks opened by contributors
+      # who do not have write permissions to the repository, but PRs
+      # via branches from the main repository should succeed.
```

Comment:
We rarely have PRs from branches in the main repository. Although possibly dependabot is an exception...

Reply:
Sorry, I noticed that this doesn't include what I mentioned above in https://github.com/sympy/sympy/pull/27183#issuecomment-2457239924 – this is also valid for PRs from maintainers/core developers Updated the comment to match with my statement in 43a123cd6e6de8391b53a8724dbcb19eee3f6834.